### PR TITLE
Hussein-dev

### PR DESCRIPTION
**Pull request by Husen-17**
please merge the following code to the main inordert to accept your previous pull request

### CODE REVIEW

File: app/Signup.jsx [hunk 1]
```diff
@@ -235,6 +279,110 @@ class Signup extends Component {
...
+  };
+  
+  createPatientObject = () => {
+    const {
+      fullName,
+      email,
+      password,
+      nationality,
+      sex,
+      //add age
```
Comment (ItisMoe) [inline app/Signup.jsx:325]: shu age not available

File: app/Signup.jsx [hunk 2]
```diff
@@ -235,6 +279,110 @@ class Signup extends Component {
...
     );
   };
+
+  //set ur own ip address
+   constructUrl = (ipAddress, endpoint) => {
+    return `http://${ipAddress}:3000${endpoint}`;
+  };
+  getAllPatients = async () => {
+    try {
+      //pass in ur ip address
```
Comment (ItisMoe) [inline app/Signup.jsx:289]: kermel shu hay?

File: app/Signup.jsx [hunk 3]
```diff
@@ -235,6 +279,110 @@ class Signup extends Component {
...
+      //age: '', 
+      // Age is not available add it
+      phoneNumber,
+      bloodType,
+      activityLevel,
+      isSmoking,
+      height: parseInt(height), // Convert height to a number
+      weight: parseInt(weight), // Convert weight to a number
+      //bloodPressure: null, 
+      //allergies: [],
```
Comment (ItisMoe) [inline app/Signup.jsx:360]: mish me7erzen 
5allina nrakkiz 3al aham

File: backend/controllers/DiseasePrediction.controller.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    "blister": 0,
+    "red_sore_around_nose": 0,
+    "yellow_crust_ooze": 0
+  };
+  
+  
+
+// Make a POST request to your Flask API
+axios.post('http://10.21.128.63:5000', requestData)
+    .then(response => {
```
Comment (ItisMoe) [inline backend/controllers/DiseasePrediction.controller.js:143]: eh handelun
terekle yehun ya3ne?!

File: package.json [hunk 5]
```diff
@@ -13,41 +13,46 @@
...
     "expo-font": "~11.10.3",
     "expo-image-picker": "~14.7.1",
     "expo-linking": "~6.2.2",
+    "expo-permissions": "^14.4.0",
     "expo-router": "~3.4.10",
     "expo-splash-screen": "~0.26.5",
     "expo-status-bar": "~1.11.1",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "openai": "^4.44.0",
```
Comment (ItisMoe) [inline package.json:37]: btw, dafa3et meshe l 7al?

**Comment by Husen-17** [comment]
for Age, the state for age is missing , make sure u are handling the state
of age
getAll patients call is just a test method for debugging purposes
for height and weight no problem keep them so
for the Disease prediction controller still working on it-> it is not
settled yet
for open AI, i saw that creating  a new accounts gets you 5 $ for free so
waiting for a new account from my family members

File: app/Signup.jsx (hunk 1 above)
Comment (Husen-17) [reply app/Signup.jsx:325]: age state is not being handled

File: app/Signup.jsx (hunk 2 above)
Comment (Husen-17) [reply app/Signup.jsx:289]: testing purposes

File: app/Signup.jsx (hunk 3 above)
Comment (Husen-17) [reply app/Signup.jsx:360]: no worries

File: backend/controllers/DiseasePrediction.controller.js (hunk 4 above)
Comment (Husen-17) [reply backend/controllers/DiseasePrediction.controller.js:143]: this method is still under development

File: package.json (hunk 5 above)
Comment (Husen-17) [reply package.json:37]: working on this issue, waiting for a new gmail account

**Pull request merged**